src/main/scala/it/bitrock/dvs/api/core/dispatcher/GlobalMessageDispatcher.scala

```diff
@@ -38,28 +41,36 @@ class GlobalMessageDispatcher(val sourceActorRef: ActorRef, kafkaPollerHub: Kafk
       forwardMessage(ApiEvent(EventType.from(e), e.asInstanceOf[EventPayload]).toJson.toString)
   }
 
+  private def maxUpdateRateToDuration(maxUpdateRate: Option[Int]): Option[FiniteDuration] =
+    for {
+      maxUpdateRate <- maxUpdateRate
+      if maxUpdateRate > 0 && 60 <= maxUpdateRate
+    } yield (60 / maxUpdateRate) seconds
+
```

Comment:
What about:
`maxUpdateRate.filter(mur => mur > 0 && 60 <= mur).map(_ / 60 seconds)`

can be more readable?

Reply:
yet another version: `maxUpdateRate.filter(1 to 60 contains _).map(60 / _ seconds)` 😄

Reply:
I'd also consider deserialising to a `FiniteDuration` instead of a `Int`. The client will send a number, but it gets interpreted as a Duration in seconds. Reasonable?

Reply:
The idea was, to get the max rate per minute that the client can handle, and don't overcome that limit. In this case, when the client sets a limit to 60 msg per minute, I can send him a message every 30 seconds(because we don't have any update for example). 

In case when we accept a duration from the client it becomes more flexible and we have to send a message after the interval in case if nothing changed also.

What do you think?

Reply:
`mur mur` doesn't look super readable to me :) Can refactor it with a filter if we agree.

Reply:
> `mur mur` doesn't look super readable to me :) Can refactor it with a filter if we agree.

It was clashing with parameter name in method, double filter is fine to me however.

Reply:
No this case after refactoring